core/src/main/java/org/testcontainers/containers/JibDockerClient.java

```diff
@@ -0,0 +1,82 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.command.InspectImageResponse;
+import com.github.dockerjava.api.command.LoadImageCallback;
+import com.google.cloud.tools.jib.api.DockerClient;
+import com.google.cloud.tools.jib.api.ImageDetails;
+import com.google.cloud.tools.jib.api.ImageReference;
+import com.google.cloud.tools.jib.http.NotifyingOutputStream;
+import com.google.cloud.tools.jib.image.ImageTarball;
+import com.google.common.io.ByteStreams;
+import lombok.Cleanup;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.images.RemoteDockerImage;
+import org.testcontainers.utility.DockerImageName;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class JibDockerClient implements DockerClient {
+
+    private static JibDockerClient instance;
+
+    private final com.github.dockerjava.api.DockerClient dockerClient = DockerClientFactory.lazyClient();
+
+    public static JibDockerClient instance() {
+        if (instance == null) {
+            instance = new JibDockerClient();
+        }
+
+        return instance;
+    }
+
+    @Override
+    public boolean supported(Map<String, String> map) {
+        return false;
+    }
+
+    @Override
+    public String load(ImageTarball imageTarball, Consumer<Long> writtenByteCountListener) throws IOException {
+        @Cleanup
+        PipedInputStream in = new PipedInputStream();
+        @Cleanup
+        PipedOutputStream out = new PipedOutputStream(in);
+        LoadImageCallback loadImage = this.dockerClient.loadImageAsyncCmd(in).exec(new LoadImageCallback());
+
+        try (NotifyingOutputStream stdin = new NotifyingOutputStream(out, writtenByteCountListener)) {
+            imageTarball.writeTo(stdin);
+        }
+
+        return loadImage.awaitMessage();
+    }
+
+    @Override
+    public void save(ImageReference imageReference, Path outputPath, Consumer<Long> writtenByteCountListener)
+        throws IOException {
+        InputStream inputStream = this.dockerClient.saveImageCmd(imageReference.toString()).exec();
```

Comment:
I think https://github.com/testcontainers/testcontainers-java/pull/5623#discussion_r934976478 also holds true here and would similarly require a `saveImageAsyncCmd` if we want to implement it cleanly.

Reply:
I know what to do tomorrow :)

Reply:
Do we still have to tackle this before merging, or can we do it as a future improvement?

Reply:
We will mark this with a `TODO` and improve it in a follow up. We don't expect this to be a blocker for early adopters.